core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/flatten.kt

```diff
@@ -5,18 +5,58 @@ import org.jetbrains.kotlinx.dataframe.DataFrame
 import org.jetbrains.kotlinx.dataframe.annotations.AccessApiOverload
 import org.jetbrains.kotlinx.dataframe.annotations.Interpretable
 import org.jetbrains.kotlinx.dataframe.annotations.Refine
+import org.jetbrains.kotlinx.dataframe.api.Select.SelectSelectingOptions
 import org.jetbrains.kotlinx.dataframe.columns.ColumnReference
 import org.jetbrains.kotlinx.dataframe.columns.toColumnSet
+import org.jetbrains.kotlinx.dataframe.documentation.DocumentationUrls
+import org.jetbrains.kotlinx.dataframe.documentation.DocumentationUrls.Distinct
+import org.jetbrains.kotlinx.dataframe.documentation.ExcludeFromSources
 import org.jetbrains.kotlinx.dataframe.impl.api.flattenImpl
 import kotlin.reflect.KProperty
 
 // region DataFrame
 
+/**
+ * ## The Flatten Operation
+ *
+ * Flattens the {@get GROUPS} column groups in the DataFrame, replacing them with their leaf columns.
+ *
+ * __NOTE:__ Columns after flattening will keep their original names.
+ * Potential column name clashes are resolved by adding minimal possible name prefix from ancestor columns.
+ *
+ * @param [keepParentNameForColumns] If true, retains the parent column name as a prefix for the flattened columns.
+ * The prefix is separated from the original column names using the provided separator.
+ * Defaults to `false`.
+ * @param [separator] The string used to separate parent column names and the original column names when `keepParentNameForColumns` is `true`.
+ * Defaults to `"_"`.
+ * {@get [FLATTEN_PARAM] @param [columns]
+ * The names of the columns or selector determining which column groups should be flattened.}
+ *
+ * @return A new [DataFrame] with the {@get GROUPS} column groups flattened.
+ *
+ * @see [Selecting Columns][SelectSelectingOptions].
+ * @see {@include [DocumentationUrls.Flatten]}
+ */
+@ExcludeFromSources
+@Suppress("ClassName")
+private interface FlattenDocs {
+    interface FLATTEN_PARAM
+}
+
+/**
+ * {@include [FlattenDocs]}
+ * {@set GROUPS all }.
```

Comment:
spaces at the end of arguments are removed by KoDEx, so you can remove the space before '}' if you like.

Also why is there a period after it?

Reply:
magic, thanks! (copy-pasted a template from distinct)